src/ralph/backends/data/fs.py

```diff
@@ -0,0 +1,274 @@
+            details (bool): Get detailed file information instead of just file paths.
+            new (bool): Given the history, list only not already read files.
+
+        Yields:
+            str: The next file path. (If details is False).
+            dict: The next file details. (If details is True).
+        """
+
+        target = Path(target) if target else self.default_directory
+        paths = [str(path) for path in target.iterdir()]
+
+        logger.debug("Found %d files", len(paths))
+
+        if new:
+            paths = set(paths) - set(self.get_command_history(self.name, "fetch"))
+            logger.debug("New files: %d", len(paths))
+
+        if not details:
+            for path in paths:
+                yield path
+
+            return
+
+        for path in paths:
+            path = Path(path)
+            stats = path.stat()
+            modified_at = datetime.fromtimestamp(int(stats.st_mtime), tz=timezone.utc)
+            yield {
+                "filename": path.name,
+                "size": stats.st_size,
+                "modified_at": modified_at.isoformat(),
+            }
+
+    @enforce_query_checks
+    def read(
+        self,
+        *,
+        query: Union[str, BaseQuery] = None,
+        target: str = None,
+        chunk_size: Union[int, None] = None,
+        raw_output: bool = False,
+    ) -> Generator[None, Union[bytes, dict], None]:
+        """Read files matching the query in the target folder and yield them.
+
+        Args:
+            query: (str or BaseQuery): The relative pattern for the files to read.
+            target (str or None): The target directory path containing the files.
+                If target is `None`, the `default_directory_path` is used instead.
+            chunk_size (int or None): The chunk size for reading files.
+
+        Yields:
+            bytes: The next chunk of the read files if `raw_output` is True.
+            dict: The next JSON parsed line of the read files if `raw_output` is False.
+        """
+
+        if not query.query_string:
+            query.query_string = self.default_query_string
+
+        if not chunk_size:
+            chunk_size = self.default_chunk_size
+
+        target = Path(target) if target else self.default_directory
+        paths = list(target.glob(query.query_string))
+
+        if not paths:
+            logger.info("No file found for query: %s", target / query.query_string)
+            return
+
+        logger.debug("Reading matching files: %s", paths)
+
+        for path in paths:
+            with path.open("rb") as file:
+                reader = self._read_raw if raw_output else self._read_dict
+                for chunk in reader(file, chunk_size):
+                    yield chunk
+
+            # The file has been read, add a new entry to the history.
+            self.append_to_history(
+                {
+                    "backend": self.name,
+                    "command": "fetch",
```

Comment:
Should we switch the history to a backend logic instead of a CLI driven approach?

```suggestion
                    "action": "read",
```

Reply:
Great Idea) Thanks) Updated this part) had to change the current HistoryMixin a bit to work with `action` too)